Incorrect handling of unnamed bit fields

Project: [libbpf](https://github.com/libbpf/libbpf)
File: `libbpf/src/netlink.c`
Function: `bpf_xdp_attach`
Error after generating test:
```cpp
/home/utbot/tmp/utbot-RWJ7/libbpf/netlink_klee.c:28:32: error: invalid operands to binary expression ('struct bpf_xdp_attach_opts' and 'int')
    klee_prefer_cex(&opts, opts<= 10);
```
Generated klee file:
```c
//////////////////////////////////////////// 
    struct bpf_xdp_attach_opts opts;
    klee_make_symbolic(&opts, sizeof(opts), "opts");
    klee_prefer_cex(&opts, opts.sz<= 10);
    klee_prefer_cex(&opts, opts.old_prog_fd >= -10  & opts.old_prog_fd <= 10);
    klee_prefer_cex(&opts, opts<= 10);
//////////////////////////////////////////// 
```
Declaration of struct:
```c
struct bpf_xdp_attach_opts {
	size_t sz;
	int old_prog_fd;
	size_t :0;
};
```

--- a/server/src/types/TypesResolver.cpp
+++ b/server/src/types/TypesResolver.cpp
@@ -90,6 +90,9 @@ void TypesResolver::resolveStruct(const clang::RecordDecl *D, const std::string 
     std::vector<types::Field> fields;
     fs::path sourceFilePath = sourceManager.getFileEntryForID(sourceManager.getMainFileID())->tryGetRealPathName().str();
     for (const clang::FieldDecl *F : D->fields()) {
+        if (F->isUnnamedBitfield()) {
+            continue;
+        }
         types::Field field;
         field.name = F->getNameAsString();
         const clang::QualType paramType = F->getType().getCanonicalType();
@@ -239,6 +242,9 @@ void TypesResolver::resolveUnion(const clang::RecordDecl *D, const std::string &
     std::vector<types::Field> fields;
     unionInfo.hasUnnamedFields = false;
     for (const clang::FieldDecl *F : D->fields()) {
+        if (F->isUnnamedBitfield()) {
+            continue;
+        }
         types::Field field;
         string fieldName = F->getNameAsString();
         field.name = fieldName;

--- a/server/test/framework/Regression_Tests.cpp
+++ b/server/test/framework/Regression_Tests.cpp
@@ -189,4 +189,19 @@ namespace {
                 } }),
             "ret");
     }
+
+    TEST_F(Regression_Test, Unnamed_Bit_Field) {
+        fs::path source = getTestFilePath("PR124.c");
+        auto [testGen, status] = createTestForFunction(source, 12);
+
+        ASSERT_TRUE(status.ok()) << status.error_message();
+
+        checkTestCasePredicates(
+            testGen.tests.at(source).methods.begin().value().testCases,
+            vector<TestCasePredicate>(
+                { [](const tests::Tests::MethodTestCase &testCase) {
+                    return !testCase.isError();
+                } }),
+            "bpf_xdp_attach");
+    }
 }

--- a/server/test/suites/regression/CMakeLists.txt
+++ b/server/test/suites/regression/CMakeLists.txt
@@ -26,4 +26,6 @@ add_library(
 add_library(PR120 PR120.c)
 set_property(TARGET PR120 PROPERTY C_STANDARD 90)
 
+add_library(PR122 PR124.c)
+
 set_target_properties(regression PROPERTIES LINK_WHAT_YOU_USE TRUE)

--- a/server/test/suites/regression/PR124.c
+++ b/server/test/suites/regression/PR124.c
@@ -1,0 +1,12 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2012-2021. All rights reserved.
+ */
+
+struct bpf_xdp_attach_opts {
+    int sz;
+    int : 0;
+};
+
+int bpf_xdp_attach(const struct bpf_xdp_attach_opts *opts) {
+    return 0;
+}

